server/src/main/java/org/apache/iotdb/db/qp/physical/crud/BatchInsertPlan.java

```diff
@@ -288,6 +291,19 @@ public long getMaxTime() {
     return maxTime;
   }
 
+  public TimeValuePair composeLastTimeValuePair(int measurementIndex) {
+    long maxTime = Long.MIN_VALUE;
+    int maxIndex = 0;
+    for (int i = 0; i < times.length; i++) {
+      if (times[i] > maxTime) {
+        maxTime = times[i];
+        maxIndex = i;
+      }
+    }
```

Comment:
No matter what measurementIndex is, this calculation process is the same. It only needs to be called once.
You can extract that method outside the the calling loop.

Reply:
Fixed